File: functions/src/dto/Room.ts
export class Room {
  rid: string;
  title: string;
  manager: string;
  maxUser: number;
  currentUser: number;
  state: string;

  constructor(
    rid: string,
    title: string,
    manager: string,
    maxUser: number,
    currentUser: number,
    state: string,
  ) {
    this.rid = rid;
    this.title = title;
    this.manager = manager;
    if (maxUser == null || undefined) {
      this.maxUser = 5;
    } else {
      this.maxUser = maxUser;
    }
    if (maxUser == null || undefined) {
      this.currentUser = 1;
    } else {
      this.currentUser = currentUser;
    }
    this.state = state;
  }
}

File: ChainCatch/app/src/main/java/com/dtd/chaincatch/home/model/dto/Room.kt
package com.dtd.chaincatch.home.model.dto

data class Room(
  val rid: String,
  val title: String,
  val manager: String,
  val maxUser: Long = 5,
  val currentUser: Long = 1,
) {
  override fun equals(other: Any?): Boolean {
    if (other is Room) return rid == other.rid
    return false
  }

  override fun hashCode(): Int {
    return super.hashCode()
  }
}
